ui/text_displayer.js

```diff
@@ -359,11 +357,9 @@ shaka.ui.TextDisplayer = class {
     captionsStyle.direction = cue.direction;
 
     // Add cue content
-    const content = this.buildCueElements_(cue.payload);
-    if (content) {
-      for (let i = 0; i < content.length; i++) {
-        captions.appendChild(content[i]);
-      }
+    const tokens = this.tokenizeCueString_(cue.payload);
+    for (const element of this.buildCueElements_(tokens)) {
```

Comment:
The tokenization and DOM-building process seems sensitive enough to warrant some automated tests.  Something that checks the hierarchies built by certain VTT payloads would be really nice to have.

You could add something to `test/ui/`.  Make sure the file ends with `_unit.js` to have it picked up automatically by the test runner.  Your tests could create a new instance of `shaka.ui.TextDisplayer` with a fake video element (`shaka.test.FakeVideo`) and a detached div (`document.createElement('div')` without appending it to the document).  You call `displayer.append([cue])` with a cue starting at time 0, `await shaka.test.Util.delay(0.5)` to let the displayer's timer fire and update the DOM, then check the structure of your detached div to see that it's correct.

Those tests would give us a good base to build on if we add voice or class support later.

What do you think?

Reply:
Tests were added in 135f116. They cover well, but this is my first time using this testing framework so I'm unsure of the acceptability of my structure/formatting.

I'm also feeling odd about:
https://github.com/google/shaka-player/blob/135f11629f5c7de82e830415984cf9ee307ec47e/test/ui/vtt_text_renderer_unit.js#L24-L28
I was fighting the compiler here a bit. Although it works, I'm sure there's a cleaner way to get that inner element out of the tree without triggering `Using properties of unknown types is not allowed`.